spring-rabbit/src/main/java/org/springframework/amqp/rabbit/core/RabbitTemplate.java

```diff
@@ -258,13 +267,19 @@ public class RabbitTemplate extends RabbitAccessor // NOSONAR type line count
 
 	private ErrorHandler replyErrorHandler;
 
+	private boolean useChannelForCorrelation;
+
+	private boolean observationEnabled;
+
+	private RabbitTemplateObservationConvention observationConvention = new RabbitTemplateObservationConvention();
```

Comment:
@marcingrzejszczak ,

is this what you have expected?

I thought the idea was just about a general `ObservationConvention<RabbitMessageSenderContext>` and that's it...